weavetest/abci.go

```diff
@@ -135,3 +139,79 @@ func (w *WeaveRunner) InBlock(executeTx func(WeaveApp) error) bool {
 	finalHash := w.app.Commit().Data
 	return !bytes.Equal(initialHash, finalHash)
 }
+
+var _ weave.ReadOnlyKVStore = (*WeaveRunner)(nil)
+
+func (w *WeaveRunner)Get(key []byte) []byte {
+	query := w.app.Query(abci.RequestQuery{
+		Path: "/",
+		Data: key,
+	})
+	// if only the interface supported returning errors....
+	if query.Code != 0 {
+		panic(query.Log)
+	}
+	// TODO: avoid importing app
+	var value app.ResultSet
+	err := value.Unmarshal(query.Value)
+	if err != nil {
+		// oh, for an error return here...
+		panic(errors.Wrap(err, "Cannot parse values"))
```

Comment:
This method can panic in two places :see_no_evil: 

Should the interface be updated to return an error as well? If yes, a new ticket should do for now.

~Would it be better to log the error and return `nil` instead of `panic`? It seems very radical to panic because the requested value was not found or cannot be deserialized.~

On the other hand, this is `weavetest`. I think it is perfectly fine to `panic` in tests :smile_cat:

Reply:
Yeah, the kvstore interface was intended for an embedded db store, which is not supposed to error on read/write. And I am abusing it over a network connection and multiple levels of parsing.

We could add this second argument, if we really need it in the normal db case, but not just for tests

Reply:
agree. panic is 🆗 for tests